livre/core/Render/RenderBrick.cpp

```diff
@@ -105,30 +105,23 @@ void RenderBrick::getScreenCoordinates( const Frustum& frustum,
                     continue;
                 in.normalize();
 
-                const double perspCorr = 1.0 / in.w();
+                Vector3d in3( in );  
```

Comment:
At least put a comment on top of it that homogenization is done on construction time. This is so implicit.